File: app/src/main/java/com/phucnguyen/lovereminder/ImageAdapter.java
package com.phucnguyen.lovereminder;

import android.content.Context;
import android.util.SparseBooleanArray;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

import java.util.List;

public class ImageAdapter extends BaseAdapter {
    Context context;
    List<Image> images;
    private SparseBooleanArray imagesToDeleteTracker;

    @Override
    public int getCount() {
        return images.size();
    }

    public ImageAdapter(Context context, List<Image> images) {
        this.context = context;
        this.images = images;
        this.imagesToDeleteTracker = new SparseBooleanArray();
    }

    @Override
    public Object getItem(int position) {
        return images.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ImageView imageView = new ImageView(context);
        if (imagesToDeleteTracker.get(position)) {
            imageView.setScaleType(ImageView.ScaleType.CENTER_CROP);
            imageView.setLayoutParams(new ViewGroup.LayoutParams(330, 350));
            imageView.setCropToPadding(true);

            Glide.with(context)
                    .load(images.get(position).getUri())
                    .thumbnail(0.5f)
                    .into(imageView);

            imageView.setBackgroundResource(R.drawable.view_border);
        } else {
            Glide.with(context)
                    .load(images.get(position).getUri())
                    .into(imageView);
            imageView.setScaleType(ImageView.ScaleType.CENTER_CROP);
            imageView.setLayoutParams(new ViewGroup.LayoutParams(330, 350));
        }
        return imageView;

    }

    @Override
    public boolean isEnabled(int position) {
        return super.isEnabled(position);
    }

    @Override
    public boolean areAllItemsEnabled() {

        return true;
    }

    public void setImages(List<Image> images) {
        this.images = images;
        notifyDataSetChanged();
    }

    public boolean toggleImagePositionToDelete(int position) {
        boolean result;
        if (imagesToDeleteTracker.get(position)) {
            result = false;
            imagesToDeleteTracker.delete(position);
        } else {
            result = true;
            imagesToDeleteTracker.put(position, true);
        }
        notifyDataSetChanged();
        return result;
    }

    public SparseBooleanArray getImagesToDelete() {
        return imagesToDeleteTracker;
    }

    public void resetTracker() {
        imagesToDeleteTracker.clear();
    }
}

File: app/src/main/java/com/phucnguyen/lovereminder/ui/adapter/ImageAdapter.kt
package com.phucnguyen.lovereminder.ui.adapter

import android.content.Context
import android.util.SparseBooleanArray
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.phucnguyen.lovereminder.R
import com.phucnguyen.lovereminder.databinding.ItemPictureBinding
import com.phucnguyen.lovereminder.model.Image

class ImageAdapter(var context: Context) : RecyclerView.Adapter<ImageAdapter.ViewHolder>() {
    interface Listener {
        fun onItemClicked(position: Int)
        fun onItemLongClicked(position: Int)
    }

     var images: List<Image>? = null
        set(value) {
            field = value
            notifyDataSetChanged()
        }
    var listener: Listener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemPictureBinding.inflate(LayoutInflater.from(context))
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        images?.let {
            val data = it[position]
            holder.bind(data)
        }

        holder.itemView.setOnClickListener {
            listener?.onItemClicked(position)
        }
        holder.itemView.setOnLongClickListener {
            listener?.let {
                it.onItemLongClicked(position)
                true
            } ?: false
        }
    }

    override fun getItemCount(): Int {
        return images?.size ?: 0
    }

    class ViewHolder(private val binding: ItemPictureBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(image: Image) {
            Glide.with(binding.ivPicture)
                .load(image.uri)
                .centerCrop()
                .into(binding.ivPicture)

            if (image.isPendingDelete) {
                binding.ivPendingTrack.visibility = View.VISIBLE
                binding.ivPicture.alpha = 0.8f
            } else {
                binding.ivPendingTrack.visibility = View.GONE
                binding.ivPicture.alpha = 1f
            }
        }
    }
}